### PR TITLE
Fixed processing of lists in config file

### PR DESCRIPTION
**Pull request by reu-res**
The lists in config file were not processed, for example
```
[context]
terminal=['tmux', 'splitw', '-h']
```

### CODE REVIEW

File: pwnlib/context/__init__.py [hunk 1]
```diff
@@ -351,7 +351,7 @@ class ContextType(object):
         'os': 'linux',
         'proxy': None,
         'signed': False,
-        'terminal': None,
+        'terminal': [],
```
Comment (zachriggle) [inline pwnlib/context/__init__.py]: This is a list instance, which may lead to weird things happening by modifying `context.terminal` in-place.  Please revert this change.

File: pwnlib/context/__init__.py [hunk 2]
```diff
@@ -1359,7 +1359,7 @@ def update_context_defaults(section):
         if key not in ContextType.defaults:
             log.warn("Unknown configuration option %r in section %r" % (key, 'context'))
             continue
-        if isinstance(ContextType.defaults[key], (str, unicode)):
+        if isinstance(ContextType.defaults[key], (str, unicode, list)):
```
Comment (zachriggle) [inline pwnlib/context/__init__.py]: This is wrong.  You cannot `saveeval.expr(...)` on a `list`.

**Comment by zachriggle** [comment]
This Pull Request is wrong.

**Comment by zachriggle** [comment]
Disregard my earlier comments, I just woke up and wasn't looking close enough.

This is a good PR if you change your usage of `list` and `[]` to `tuple` and `tuple()`.

**Comment by zachriggle** [comment]
It would be excellent if we could also add tests for this functionality, clearly it's necessary.

**Comment by TethysSvensson** [comment]
I am not sure of an obvious way to test this
